syntax = "proto3";

package ml;
option go_package="./services";
// The request message containing a single image.
message ImageRequest {
  bytes image = 1;
}

// The response message containing the classification results for a single image.
message ImageResponse {
  repeated string classifications = 1;
}

// The image classification service definition.
service MLService {
  // Bidirectional streaming RPC method for image classification.
  rpc DetectObjects (ImageRequest) returns (ImageResponse);
}
